columnflow/weight/__init__.py

```diff
@@ -81,13 +81,15 @@ def update_cls_dict(cls_name, cls_dict, get_attr):
                     raise Exception(
                         f"weight producer {cls_name} received both mc_only and data_only",
                     )
+
                 if mc_only or data_only:
                     if cls_dict.get("skip_func"):
                         raise Exception(
                             f"weight producer {cls_name} received custom skip_func, but either "
                             "mc_only or data_only are set",
                         )
 
+                if "skip_func" not in cls_dict:
```

Comment:
I had to think about the implications here a bit more, but I think this is what we want, i.e.

- complain if `mc_only` or `data_only` **and** a `skip_func` is set, and then
- create a `skip_func` if none is defined so far.

We should also add this subtle change to the other decorators that offer a `skip_func`. I can do that @dsavoiu and amend this PR, unless you want to go ahead :+1:

Reply:
Added this just now in 81ca127 for consistency with calibrators, selectors and producers.